torch/ao/quantization/fx/prepare.py

```diff
@@ -475,6 +512,43 @@ def _get_arg_target_dtype_as_output(
     # TODO: should support is_dynamic here as well
     return output_act_dtype
 
+def _populate_arg_to_ctr_for_args(
+    arg_or_args: Union[Argument, Tuple[Argument, ...]],
+    input_obs_or_fq_ctr_for_args: Optional[Tuple[Any]],
+    arg_to_ctr: Dict[Argument, Any],
+) -> None:
+    if not isinstance(input_obs_or_fq_ctr_for_args, tuple):
+        # if input_obs_or_fq_ctr_for_args is a single element,
+        # use the same obs_or_fq_ctr for all args
+        if isinstance(arg_or_args, (tuple, list)):
+            for inner_arg in arg_or_args:
+                _populate_arg_to_ctr_for_args(inner_arg, input_obs_or_fq_ctr_for_args, arg_to_ctr)
+        elif isinstance(arg_or_args, Node):
+            arg_to_ctr[arg_or_args] = input_obs_or_fq_ctr_for_args
+    else:
+        for arg, input_obs_or_fq_ctr in zip(arg_or_args, input_obs_or_fq_ctr_for_args):
```

Comment:
Also, how we gurantee that coming here means both `arg_or_args` and `input_obs_or_fq_ctr_for_args` have the same structure. I think we should enforce that here such that we ask for the signatures to match.

Reply:
I see you address this later